test/steps/kogitoexplainability.go

```diff
@@ -0,0 +1,128 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/cucumber/godog"
+	"github.com/kiegroup/kogito-cloud-operator/test/framework"
+	"github.com/kiegroup/kogito-cloud-operator/test/steps/mappers"
+	bddtypes "github.com/kiegroup/kogito-cloud-operator/test/types"
+)
+
+/*
+	DataTable for Explainability:
+	| runtime-request | cpu/memory  | value                     |
+	| runtime-limit   | cpu/memory  | value                     |
+	| runtime-env     | varName     | varValue                  |
+	| kafka           | externalURI | kafka-bootstrap:9092      |
+	| kafka           | instance    | external-kafka            |
+*/
+
+func registerKogitoExplainabilityServiceSteps(ctx *godog.ScenarioContext, data *Data) {
+	ctx.Step(`^Install Kogito Explainability with (\d+) replicas$`, data.installKogitoExplainabilityServiceWithReplicas)
+	ctx.Step(`^Install Kogito Explainability with (\d+) replicas with configuration:$`, data.installKogitoExplainabilityServiceWithReplicasWithConfiguration)
+	ctx.Step(`^Kogito Explainability has (\d+) pods running within (\d+) minutes$`, data.kogitoExplainabilityHasPodsRunningWithinMinutes)
+	ctx.Step(`^Explainability result is available in the Trusty service within (\d+) minutes$`, data.explainabilityResultIsAvailable)
+}
+
+func (data *Data) installKogitoExplainabilityServiceWithReplicas(replicas int) error {
+	explainability := framework.GetKogitoExplainabilityResourceStub(data.Namespace, replicas)
+	return framework.InstallKogitoExplainabilityService(data.Namespace, framework.GetDefaultInstallerType(), &bddtypes.KogitoServiceHolder{KogitoService: explainability})
+}
+
+func (data *Data) installKogitoExplainabilityServiceWithReplicasWithConfiguration(replicas int, table *godog.Table) error {
+	explainability := &bddtypes.KogitoServiceHolder{
+		KogitoService: framework.GetKogitoExplainabilityResourceStub(data.Namespace, replicas),
+	}
+
+	if err := mappers.MapKogitoServiceTable(table, explainability); err != nil {
+		return err
+	}
+
+	return framework.InstallKogitoExplainabilityService(data.Namespace, framework.GetDefaultInstallerType(), explainability)
+}
+
+func (data *Data) kogitoExplainabilityHasPodsRunningWithinMinutes(podNb, timeoutInMin int) error {
+	return framework.WaitForKogitoExplainabilityService(data.Namespace, podNb, timeoutInMin)
+}
+
+type executionsResponse struct {
+	Executions []execution `json:"headers"`
+}
+
+type execution struct {
+	ExecutionID string `json:"executionId"`
+}
+
+func (data *Data) explainabilityResultIsAvailable(timeoutInMin int) error {
```

Comment:
If I understood well the logic in this method, we're doing the following:

1. Await for call http://trusty/executions to check whether contains DECISION
2. Call http://trusty/executions and marshall to executionsResponse
3. Get the first execution from executionsResponse
4. Await for call http://trusty/executions/decisions/{first execution ID}/explanations/saliencies to check whether the content is SUCCEEDED

I think we're doing one unnecessary call (step 2) and one unnecessary await (step 4). Also, getting the first execution can be tricky, is there a way we can ensure that the decision is the one we expect? Maybe by name?

Reply:
Hi @Sgitario , yes, the second step might be redundant. It's like asserting that an object is not null and then asserting that the properties have a particular value. 
The second question imo is not a problem, since we are executing just one request and there should be just one execution retrieved by the trusty service. We can improve checking that the size of the array is just one